projects/packages/woocommerce-analytics/src/class-woo-analytics-trait.php

```diff
@@ -396,16 +395,16 @@ public function maybe_start_session() {
 			$this->is_new_session = true;
 
 			$session_expiration = $this->get_session_expiration_time();
-			$event_js           = $this->process_event_properties( 'woocommerceanalytics_session_started' );
-			$cookie_js          = "
-            const sessionData = JSON.stringify({
-                    session_id: '$this->session_id',
-                    landing_page: encodeURIComponent('$this->landing_page'),
-                    expires: '$session_expiration',
-            });
-            document.cookie = `woocommerceanalytics_session=\${sessionData}; expires=$session_expiration; path=/; secure; samesite=strict`;
-            ";
+			$session_data       = array(
+				'session_id'   => $this->session_id,
+				'landing_page' => rawurlencode( $this->landing_page ),
+				'expires'      => $session_expiration,
+			);
+			$encoded_data       = wp_json_encode( $session_data );
+			$cookie_js          = "document.cookie = 'woocommerceanalytics_session={$encoded_data}; expires={$session_expiration}; path=/; secure; samesite=strict';";
```

Comment:
I considered using PHP to set the cookie, but I realized that it might not be the best approach for our use case since we may want to delay setting the cookie until getting user consent for tracking in the future implementation.